src/frontend/src/lib/derived/tokens.derived.ts

```diff
@@ -31,7 +32,11 @@ export const tokensToPin: Readable<TokenToPin[]> = derived(
 );
 
 export const sortedTokens: Readable<Token[]> = derived(
-	[tokens, tokensToPin, exchanges],
-	([$tokens, $tokensToPin, $exchanges]) =>
-		pinTokensAtTop({ $tokens: sortTokens({ $tokens, $exchanges }), $tokensToPin })
+	[tokens, tokensToPin, exchanges, balancesStore],
+	([$tokens, $tokensToPin, $exchanges, $balancesStore]) =>
+		pinTokensWithBalanceAtTop({
```

Comment:
I'm a bit concerned about the performant. This PR introduces more `map` and `filter` and `sort` over the list of tokens and various stores. I'm sure it's fine when user have few tokens but, I fear it becomes laggy when the amount of data grows. 

For example here there is `pinTokensWithBalanceAtTop` that manipulate the list which then call `pinTokensAtTop` that manipulates the list which then use `sortTokens` that manipulate the list as well.

Is there any potential of performance improvements?

Reply:
As a first thought about it, I am thinking that ultimately we will have to have 3 sorts separately:
1. tokens with balance/usd_value
2. tokens that are forced to be pinned on top
3. all the others sorted by market_cap/name

So, the best way to reduce excessive calculation could be starting to add some props already to the tokens (as you suggested in the next comment)

Reply:
In this PR you do `.map().filter().filter()` to compute `tokensWithBalanceToPin`. We can start by using a `.reduce()` maybe?

Reply:
check last commit please